viewing_party/party.py

```diff
@@ -1,23 +1,142 @@
+                    for movie_items in item:
+                        if movie_items == "genre": #append "genre" to empty genre_list
+                            genre_list.append(item[movie_items])
+                most_popular_genre = max(set(genre_list), key = genre_list.count) #looking for most popular genre in the list
+    return most_popular_genre
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
 # -----------------------------------------
+def get_unique_watched(user_data):
+    #unique movies only by user and return list
+    watched_by_user = user_data['watched']
+    watched_by_friends = get_movies_watched_by_friends(user_data)
+    user_unique_movies = []
+    for user_movie in watched_by_user:
+        if user_movie not in watched_by_friends and user_movie not in user_unique_movies:
+                user_unique_movies.append(user_movie)
+    return user_unique_movies
+
+def get_movies_watched_by_friends(user_data):
+    #all friends whatched movies togheter and return a list
+    watched_by_friends = []
+
+    for friends in user_data['friends']:
+        for movies in friends.values():
+            for movie in movies:
+                watched_by_friends.append(movie)
+    return watched_by_friends
+
+def get_friends_unique_watched(user_data):
+    #look for unique movies watched only by friends and return list
+    watched_by_user = user_data['watched'] 
+    watched_by_friends = get_movies_watched_by_friends(user_data) 
+    friends_movies_unique = []
+
+    for movie in watched_by_friends:
+        if movie not in watched_by_user and movie not in friends_movies_unique:
+            friends_movies_unique.append(movie)
+    
+    return friends_movies_unique
 
         
 # -----------------------------------------
 # ------------- WAVE 4 --------------------
 # -----------------------------------------
+def get_available_recs(user_data):
+    movie_friends_watched_list = get_friends_unique_watched(user_data) # from wave 3
+    final_list = list()
+
+    for movie in movie_friends_watched_list:
+        if movie["host"] in user_data["subscriptions"]:
+            final_list.append(movie)
+    return final_list
 
 # -----------------------------------------
 # ------------- WAVE 5 --------------------
 # -----------------------------------------
 
+def get_new_rec_by_genre(user_data):
+    
+    recommended_movies = []
+    friends_watched = get_friends_unique_watched(user_data) # from wave 3
+    most_watched_genre = get_most_watched_genre(user_data) # from wave 2
+
+    for movie in friends_watched:
+        if movie["genre"] == most_watched_genre:
+            recommended_movies.append(movie)
+    return recommended_movies
+
+def get_rec_from_favorites(user_data):
+    recommended_movies = []
+    friends_titles_watched = []
+
+    for friend in user_data["friends"]:
+        for movie in friend["watched"]:
+            friends_titles_watched.append(movie["title"])

```

Comment:
Given that the  `get_friends_unique_watched` function from wave 3 does exactly this but with the whole movie, there's no real need for this `for` loop, you could have just did something similar as you did in `get_new_rec_by_genre` above:

```python
recommended_movies = []
friends_watched = get_friends_unique_watched(user_data)

for movie in user_data["favorites"]:
    if movie not in friends_watched:
        recommended_movies.append(movie)

return recommended_movies